integration/query_evalution_test.go

```diff
@@ -0,0 +1,401 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+func TestEvalutionMod(t *testing.T) {
+	t.Parallel()
+	ctx, collection := setup(t)
+
+	_, err := collection.InsertMany(ctx, []any{
+		bson.D{{"_id", "Zero"}, {"value", 0}},
+		bson.D{{"_id", "NegativeZero"}, {"value", math.Copysign(0, -1)}},
+		bson.D{{"_id", "Int32_1"}, {"value", int32(4080)}},
+		bson.D{{"_id", "Int32_2"}, {"value", int32(1048560)}},
+		bson.D{{"_id", "Int32_3"}, {"value", int32(268435440)}},
+		bson.D{{"_id", "Int64_1"}, {"value", int64(1099511628000)}},
+		bson.D{{"_id", "Int64_2"}, {"value", int64(281474976700000)}},
+		bson.D{{"_id", "Int64_3"}, {"value", int64(72057594040000000)}},
+		bson.D{{"_id", "Nil"}, {"value", nil}},
+		bson.D{{"_id", "String"}, {"value", "12"}},
+		bson.D{{"_id", "NaN"}, {"value", math.NaN()}},
+		bson.D{{"_id", "Infinity"}, {"value", math.Inf(0)}},
+		bson.D{{"_id", "InfinityNegative"}, {"value", math.Inf(-1)}},
+		bson.D{{"_id", "InfinityPositive"}, {"value", math.Inf(+1)}},
+		bson.D{{"_id", "SmallestNonzeroFloat64"}, {"value", math.SmallestNonzeroFloat64}},
+		bson.D{{"_id", "PositiveNumber"}, {"value", 123456789}},
+		bson.D{{"_id", "NegativeNumber"}, {"value", -123456789}},
+		bson.D{{"_id", "MaxInt64"}, {"value", math.MaxInt64}},
+		bson.D{{"_id", "MaxInt64_float"}, {"value", float64(math.MaxInt64)}},
+		bson.D{{"_id", "MaxInt64_plus"}, {"value", float64(math.MaxInt64 + 1)}},
+		bson.D{{"_id", "MaxInt64_overflowVerge"}, {"value", 9.223372036854776832e+18}},
+		bson.D{{"_id", "MaxInt64_overflow"}, {"value", 9.223372036854776833e+18}},
+		bson.D{{"_id", "MaxFloat64_minus"}, {"value", 1.79769e+307}},
+		bson.D{{"_id", "MaxFloat64"}, {"value", math.MaxFloat64}},
+		bson.D{{"_id", "MinInt64"}, {"value", math.MinInt64}},
+		bson.D{{"_id", "MinInt64_float"}, {"value", float64(math.MinInt64)}},
+		bson.D{{"_id", "MinInt64_minus"}, {"value", float64(math.MinInt64 - 1)}},
+		bson.D{{"_id", "MinInt64_overflowVerge"}, {"value", -9.223372036854776832e+18}},
+		bson.D{{"_id", "MinInt64_overflow"}, {"value", -9.223372036854776833e+18}},
+	})
+	require.NoError(t, err)
+
+	for name, tc := range map[string]struct {
+		q           bson.D
```

Comment:
would you be so kind to rename `q` to `filter`  to match other places where we use this kind of entity? Thank you.